autogpt_platform/frontend/src/components/Flow.tsx

```diff
@@ -656,6 +655,30 @@ const FlowEditor: React.FC<{
     },
   ];
 
+  const handleRunButton = useCallback(async () => {
+    if (isRunning) return;
+    if (!savedAgent) {
+      toast({
+        title: `Please save the agent first, using the button in the left sidebar.`,
+      });
+      return;
+    }
+    await saveAgent();
```

Comment:
To make *this* work I had to rewrite most of `useAgentGraph` so it returns a promise we can wait for